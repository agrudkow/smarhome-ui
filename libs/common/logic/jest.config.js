module.exports = {
  name: 'common-logic',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/common/logic',
};
