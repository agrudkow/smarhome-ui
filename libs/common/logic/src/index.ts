export * from './lib/use-accordion';
export * from './lib/use-window-dimensions';
