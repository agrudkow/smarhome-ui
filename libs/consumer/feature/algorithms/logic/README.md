# consumer-feature-algorithms-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-algorithms-logic` to execute the unit tests via [Jest](https://jestjs.io).
