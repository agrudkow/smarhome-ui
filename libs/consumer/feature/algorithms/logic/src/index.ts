export * from './lib/algorithm.interface';
export * from './lib/algorithms-cells-parser';
export * from './lib/algorithms-data-parser';
