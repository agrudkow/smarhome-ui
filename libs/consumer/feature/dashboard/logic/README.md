# consumer-feature-dashboard-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-dashboard-logic` to execute the unit tests via [Jest](https://jestjs.io).
