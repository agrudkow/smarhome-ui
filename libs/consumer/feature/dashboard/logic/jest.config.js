module.exports = {
  name: 'consumer-feature-dashboard-logic',
  preset: '../../../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory:
    '../../../../../coverage/libs/consumer/feature/dashboard/logic',
};
