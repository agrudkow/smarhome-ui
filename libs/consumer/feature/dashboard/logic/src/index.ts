export * from './lib/consumer-feature-dashboard-logic';
