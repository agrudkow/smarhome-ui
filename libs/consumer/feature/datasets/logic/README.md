# consumer-feature-datasets-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-datasets-logic` to execute the unit tests via [Jest](https://jestjs.io).
