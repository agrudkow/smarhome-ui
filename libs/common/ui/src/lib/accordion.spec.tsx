import React from 'react';
import { customRender } from '@smarthome/common/logic';
import Accordion from './accordion';

describe(' Accordion', () => {
  it('should render successfully', () => {
    const { baseElement } = customRender(
      <Accordion summary="" onChange={jest.fn()} expanded={true} />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot when expanded', () => {
    const { baseElement } = customRender(
      <Accordion summary="" onChange={jest.fn()} expanded={true} />
    );
    expect(baseElement.firstChild).toMatchSnapshot();
  });

  it('should match snapshot when collapsed', () => {
    const { baseElement } = customRender(
      <Accordion summary="" onChange={jest.fn()} expanded={true} />
    );
    expect(baseElement.firstChild).toMatchSnapshot();
  });
});
