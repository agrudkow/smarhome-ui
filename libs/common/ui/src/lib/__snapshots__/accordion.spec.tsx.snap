// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Accordion should match snapshot when collapsed 1`] = `
.c0 {
  margin: 5px 0;
  background-color: #fff;
  border: 1px solid #1BBDCE;
  border-radius: 6px;
}

<div>
  <div
    class="MuiPaper-root MuiAccordion-root c0 Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content Mui-expanded"
      />
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-entered"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="MuiAccordionDetails-root"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[` Accordion should match snapshot when expanded 1`] = `
.c0 {
  margin: 5px 0;
  background-color: #fff;
  border: 1px solid #1BBDCE;
  border-radius: 6px;
}

<div>
  <div
    class="MuiPaper-root MuiAccordion-root c0 Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content Mui-expanded"
      />
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-entered"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="MuiAccordionDetails-root"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
